apps/web/src/pages/Pools/IndexPoolPage/components/helpers.tsx

```diff
@@ -135,3 +147,118 @@ export function ownerNameAccessor(pool: Pool) {
 export function ownerEmailAccessor(pool: Pool) {
   return pool.owner && pool.owner.email ? pool.owner.email.toLowerCase() : "";
 }
+
+type TransformPoolInputArgs = {
+  search: SearchState;
+};
+
+export function transformPoolInput({ search }: TransformPoolInputArgs) {
+  if (
+    typeof search.term === "undefined" &&
+    typeof search.type === "undefined"
+  ) {
+    return undefined;
+  }
+
+  return {
+    generalSearch: !!search.term && !search.type ? search.term : undefined,
+    name: search.type === "name" ? search.term : undefined,
+    ownerName: search.type === "ownerName" ? search.term : undefined,
+    ownerEmail: search.type === "ownerEmail" ? search.term : undefined,
+    team: search.type === "type" ? search.term : undefined,
+  };
+}
+
+export function getOrderByClause(
```

Comment:
Setting up this sorting is so much more complicated than I remembered 🫣

Reply:
Oh yeah, thankfully we had an example I could copy and paste from :stuck_out_tongue: